api/v1alpha1/validation.go

```diff
@@ -0,0 +1,51 @@
+package v1alpha1
+
+import (
+	"fmt"
+
+	"github.com/go-playground/validator"
+)
+
+// use a single instance of MetadataValidator as per docs because it
+// caches struct info
+var metadataValidator *validator.Validate
+
+// getMetadataValidator - returns the MetadataValidator Singleton
+func getMetadataValidator() *validator.Validate {
+	if metadataValidator != nil {
+		return metadataValidator
+	}
+
+	// instantiate the singleton
+	metadataValidator = validator.New()
+
+	// register validation functions here
+	// TODO
+	// ...
+
+	return metadataValidator
+}
+
+// Validate - abstracts details behind AddonMetadata validation
+// returns pointer to list of FieldError so we can nil the result
+func (a *AddonMetadata) Validate() *[]validator.FieldError {
+	metadataValidator = getMetadataValidator()
+
+	if err := metadataValidator.Struct(a); err != nil {
+		res := []validator.FieldError{}
+		for _, fieldError := range err.(validator.ValidationErrors) {
+			res = append(res, fieldError)
+		}
+		return &res
+	}
+
+	// no errors
+	return nil
+}
+
+// PrintValidationErrors - helper to pretty print validationErrors and fail
+func PrintValidationErrors(fieldErrors *[]validator.FieldError) {
+	for _, err := range *fieldErrors {
+		fmt.Println(err)
+	}
+}
```

Comment:
Does it make more sense to move this to a separate utils package?

Reply:
done